### PR TITLE
Travis use focal for apt based v3

### LINKED ISSUE #10902: TST: s390x and arm64 cron jobs fail after switch to travis-ci.com
### Description

Since travis-ci.org is going away, we had no choice but to switch to travis-ci.com as instructed. However, after the switch, `s390x` and `arm64` cron jobs started to fail.

Would be nice if we can get this sorted out before release to ensure that no real failures have crept in.

### s390x

Example log: https://travis-ci.com/github/astropy/astropy/jobs/402407537
```
...
Processing triggers for ccache (3.4.1-1) ...
Updating symlinks in /usr/lib/ccache ...
E: Could not configure 'libc6:s390x'. 
E: Could not perform immediate configuration on 'libnss-nis:s390x'. Please see man 5 apt.conf under APT::Immediate-Configure for details. (2)
The command "if [ $SETUP_METHOD == 'apt' ]; then curl https://ftp-master.debian.org/keys/archive-key-10.asc | sudo apt-key add -; echo "deb http://ftp.us.debian.org/debian testing main" | sudo tee -a /etc/apt/sources.list; sudo apt-get -qq update; sudo apt-get install -t testing -y --no-install-recommends ${APT_DEPENDENCIES}; fi" failed and exited with 100 during .
```

### arm64

Example log: https://travis-ci.com/github/astropy/astropy/jobs/402407538

```
...
E: Failed to fetch http://ftp.us.debian.org/debian/pool/main/t/twisted/python3-twisted-bin_18.9.0-11+b1_arm64.deb  Undetermined Error [IP: 208.80.154.15 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
Fetched 137 MB in 16s (8620 kB/s)
The command "if [ $SETUP_METHOD == 'apt' ]; then curl https://ftp-master.debian.org/keys/archive-key-10.asc | sudo apt-key add -; echo "deb http://ftp.us.debian.org/debian testing main" | sudo tee -a /etc/apt/sources.list; sudo apt-get -qq update; sudo apt-get install -t testing -y --no-install-recommends ${APT_DEPENDENCIES}; fi" failed and exited with 100 during .
```

### PR DESCRIPTION
**Pull request by mhvk**
fixes #10902 ~(maybe)~

trying to use more of the standard travis things, so there is less need to override with Debian testing (or perhaps pip)

### CODE REVIEW

**Comment by mhvk** [comment]
OK, I think I got the `s390` and `arm` builds going again, following @olebole's advice to use the more recent ubuntu that has become available. It is now considerably simpler, which I think is great.

In order to get tests to pass, however, I had to ignore a deprecation message (because `doctestplus` is too old, I think), and, more worryingly, comment out a checksum test in `io.fits`, one that was already failing on windows and now failed on (big-endian) s390 as well (since @embray introduced it, pinging...). My sense would be to raise a separate issue about that - something like a checksum test should pass on all platforms, or we should understand why it doesn't.

Anyway, if people agree it is OK to move forward with this, I'll make the runs part of cron again, and we can merge.

p.s. Just in case it isn't obvious, I cancelled all circle-ci and travis test runs except the two relevant ones.

**Comment by mhvk** [comment]
p.s. link to run with failed `io.fits` test: https://travis-ci.com/github/astropy/astropy/jobs/403287620 (the also failed doctest has not recurred, for some reason)

**Comment by olebole** [comment]
@mhvk Can you specify the failing checksum test? I usually run all tests during the build, however with a slightly different setup (namely, I use system libs wherever possible), and I currently didn't disable a `io.fits` test (and usually I disable them only after I opened an issue here).

**Comment by olebole** [comment]
The doctest one is documented in #9657.

**Comment by mhvk** [comment]
@olebole - thanks for the link to the issue about the doctest - on `s390`, that one only occurred once, so there is definitely some flakiness involved. 

For `io.fits`, it is `astropy.io.fits.tests.test_checksum.TestChecksumFunctions.test_compressed_image_data_float32`. Here, I am similarly surprised that it fails, since it must have passed previously (after all, `s390` was working fine). It may just be a rather out-of-date package on Focal. Nevertheless, given that the same test fails on windows too, it does not seem quite worth for it to hold up getting testing otherwise back on track.

File: .travis.yml
Comment (pllim) [inline .travis.yml:197]: I think we need to move these back to cron?

File: astropy/io/fits/tests/test_checksum.py
Comment (pllim) [inline astropy/io/fits/tests/test_checksum.py:278]: Can we still not run them for OSX and little-endian jobs? Maybe @saimn can chime in here?

File: setup.cfg
Comment (pllim) [inline setup.cfg:142]: Either add a comment on why this was added or try to grab the latest release of `pytest-doctestplus` instead.

Comment (mhvk) [reply setup.cfg:142]: OK, will do.

File: .travis.yml
Comment (mhvk) [reply .travis.yml:197]: Yes, will do that the moment the general ideas here are approved!

File: astropy/io/fits/tests/test_checksum.py
Comment (mhvk) [reply astropy/io/fits/tests/test_checksum.py:278]: I've raise a separate issue - #10921 - a test that fails on some platforms seems rather useless.

File: setup.cfg
Comment (mhvk) [reply setup.cfg:142]: p.s. I wish there was a way to tell `pip` to upgrade packages only if a new version is required by something being installed, rather than the all-or-nothing `--upgrade`.

Comment (saimn) [reply setup.cfg:142]: That's what he should do, but our requirements in pytest-astropy are not up-to-date: https://github.com/astropy/pytest-astropy/blob/master/setup.cfg#L32-L43

File: astropy/io/fits/tests/test_checksum.py
Comment (saimn) [reply astropy/io/fits/tests/test_checksum.py:278]: Hmm, the fact that it was passing before on travis.org, and that @olebole also says it is passing on the debian CI let me think that this is not an architecture issue. Maybe some floating point differences as the comment says, but it seems weird that a checksum algorithm is not robust to that (and I don't know well enough the algorithm to say anything useful).

File: .travis.yml
Comment (saimn) [inline .travis.yml:207]: Why install astropy here ? Also cython should be installed with the other build requirements by pip.

File: setup.cfg
Comment (mhvk) [reply setup.cfg:142]: Ah, I see, so I can add a comment that this can be removed once we require a more recent version of doctestplus.

File: astropy/io/fits/tests/test_checksum.py
Comment (olebole) [reply astropy/io/fits/tests/test_checksum.py:278]: @mhvk maybe one could just save the resulting fits file and copy it back for debugging?
@saimn IMO checksum [just goes over the bits](https://fits.gsfc.nasa.gov/registry/checksum/checksum) and therefore will fail by any difference.

File: .travis.yml
Comment (mhvk) [reply .travis.yml:207]: I should add a comment: the nice thing of doing that is that it automatically pulls in `numpy` and `erfa` (`wcslib` and `libcfitsio` as well, but those also get pulled in by their `-dev`) and everything else astropy needs. And with that, I can use the same tests to check that we can use system libraries (both C libraries, and via `--use-system-libraries`, python packages). I.e., I try here to let `pip` do as little as possible, with the hope that these tests will make packaging on linux systems easier (and, selfishly, my own setup!).

File: astropy/io/fits/tests/test_checksum.py
Comment (mhvk) [reply astropy/io/fits/tests/test_checksum.py:278]: Indeed, we need to debug somehow, but maybe as part of #10921? I don't think it needs to hold up this PR - we should get our CI working again...

File: .travis.yml
Comment (bsipocz) [reply .travis.yml:207]: explicit is still better than implicit, I vaguely recall it bite back to rely on this automated pull of packages rather than spelling them out (I have nothing against installing numpy/erfa/etc from the linux system packaging rather than pip)

Comment (mhvk) [reply .travis.yml:207]: @bsipocz - fair point! Happy to change. (Though what I have here also has its advantages in being realistic - at least, taking myself as an example, I do have astropy installed from the distribution while working on development...)

File: astropy/io/fits/tests/test_checksum.py
Comment (astrofrog) [reply astropy/io/fits/tests/test_checksum.py:278]: Maybe add a reference to the new issue here

Comment (mhvk) [reply astropy/io/fits/tests/test_checksum.py:278]: Done

File: .travis.yml
Comment (mhvk) [reply .travis.yml]: OK, added explicit dependencies (now testing whether I didn't mess up...)

**Comment by mhvk** [comment]
Darn, immediately proving its worth - #10360 is not safe for big-endian.  I'll try to include a fix here, since with the setup in master it cannot be tested.

**Comment by bsipocz** [comment]
> Darn, immediately proving its worth - #10360 is not safe for big-endian. I'll try to include a fix here, since with the setup in master it cannot be tested.

if it's a longer fix, I would advocate for getting this merged sooner and having the fix separately. In case any other PRs are problematic.

**Comment by mhvk** [comment]
@bsipocz - good point, let's get this one in (after a final commit that puts the two back to Cron only) and I'll just properly raise an issue - the fix is only half-obvious.

**Comment by mhvk** [comment]
@bsipocz - OK, I moved the s390 and arm setups back to Cron, so it is the same as it was before (but now works...). OK to merge?

p.s. arm test did pass, so the setup is OK: https://travis-ci.com/github/astropy/astropy/jobs/405392254

File: setup.cfg
Comment (pllim) [reply setup.cfg:142]: I don't see this comment added. Did I miss something?

Comment (mhvk) [reply setup.cfg:143]: No, I missed something, sorry!!! Now added....

**Review by pllim — APPROVED** [review]
LGTM, but should wait for @bsipocz to have another look before merge.

**Review by bsipocz — APPROVED** [review]

**Comment by bsipocz** [comment]
thanks @mhvk!

**Comment by mhvk** [comment]
Thanks all for the reviews!

**Comment by mhvk** [comment]
Hmm, sadly it looks like the solution with ignoring a deprecation warning via `setup.cfg` was very flaky - I found that it only worked some times in #10931. There, I went instead with the suggestion of @pllim to just upgrade `pytest-doctestplus` (oddly, that previously caused problems, but perhaps just because I was messing something else up).

**Comment by mhvk** [comment]
Anyway, the good news at least is that the big-endian tests immediately found a bug...

**Pull request merged**
